project-base/storefront/urql/errorExchange.ts

```diff
@@ -23,46 +29,69 @@ export const getErrorExchange =
                         return;
                     }
 
+                    if (isClient && hasFriendlyUrlQueryFailedWith500(operation, error)) {
+                        if (publicRuntimeConfig.errorMessageVerbosity === 'devel') {
+                            throw new Error(JSON.stringify(mapGraphqlErrorForDevelopment(error.graphQLErrors[0])));
+                        }
+
+                        throw new Error('Internal Server Error');
+                    }
+
                     const isAuthError = error.response?.status === 401;
                     if (isAuthError) {
                         removeTokensFromCookies(context);
 
                         return;
                     }
 
-                    const parsedErrors = t ? getUserFriendlyErrors(error, t) : undefined;
-                    if (!parsedErrors) {
-                        return;
+                    if (publicRuntimeConfig.errorMessageVerbosity === 'devel') {
+                        handleErrorMessagesForDevelopment(error);
+                    } else {
+                        handleErrorMessagesForUsers(error, t, operation);
                     }
+                }),
+            );
+        };
+    };
 
-                    if (parsedErrors.userError) {
-                        logException(parsedErrors.userError);
-                    }
+const handleErrorMessagesForDevelopment = (error: CombinedError) => {
+    logException(error);
+    error.graphQLErrors
+        .map((graphqlError) => mapGraphqlErrorForDevelopment(graphqlError))
+        .forEach((simplifiedGraphqlError) => showErrorMessage(JSON.stringify(simplifiedGraphqlError)));
+};
 
-                    const isCartError = operation.query === CartQueryDocumentApi;
-                    if (isCartError) {
-                        handleCartError(parsedErrors);
+const handleErrorMessagesForUsers = (error: CombinedError, t: Translate | undefined, operation: Operation) => {
+    const parsedErrors = t ? getUserFriendlyErrors(error, t) : undefined;
```

Comment:
Hmm, if you go higher in the tree where is this function called I see that `t` is always being sent. Can we please adjust this so we don't have unnecessary conditions like this?

Reply:
We always send it, but it can be undefined. However, I still moved the condition one level higher

Reply:
I don't see it can be undefined 🤔Maybe you haven't gone deep enough. If I go where this function is used, I found only one place 

![image](https://github.com/shopsys/shopsys/assets/11465991/3bb09246-2761-4a42-92fd-8e46e507b735)

here it is marked that it can be undefined, ok. But If I see where this function is used I found again only one place

![image](https://github.com/shopsys/shopsys/assets/11465991/fcba72a1-4f70-4e31-8223-3ed6a8b6b76a)

where it is marked as defined. So there is no reason mark it as undefined. Let's refactor this please.

Reply:
Wow, you are right. Thanks for being so persistent. I removed it now.